wp-includes/components/class-wp-service-worker-navigation-routing-component.php

```diff
@@ -298,6 +298,31 @@ public function serve( WP_Service_Worker_Scripts $scripts ) {
 		$revision .= sprintf( ';user-%d', get_current_user_id() );
 
 		if ( ! is_admin() ) {
+
+			// @todo Allow different routes to have varying caching strategies?
+
+			/**
+			 * Filters caching strategy used for frontend navigation requests.
+			 *
+			 * @since 0.2
+			 * @see WP_Service_Worker_Caching_Routes::register()
+			 *
+			 * @param string $caching_strategy Caching strategy to use for frontend navigation requests.
+			 */
+			$caching_strategy = apply_filters( 'wp_service_worker_navigation_caching_strategy', WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_ONLY );
```

Comment:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
This would be good to look at as part of iterating on the PHP architecture. /cc @felixarntz

Reply:
Haha. It seems my failed attempts last night to post this comment on GitHub when the site was down were not failures after all.

Reply:
In short, the idea is to let there be a proper method for setting the navigation caching strategy rather than just resorting to a filter.

So instead of doing:

Reply:
In short, the idea is to let there be a proper method for setting the navigation caching strategy rather than just resorting to a filter.

So instead of doing:
```php
$caching_strategy = apply_filters( 'wp_service_worker_navigation_caching_strategy', WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_ONLY );
```
There could be an explicit call from plugin code that runs in the `wp_front_service_worker` action, like so:
```php
add_action( 'wp_front_service_worker', function( $scripts ) {
    $scripts->navigation_routing->set_caching_strategy( WP_Service_Worker_Caching_Routes::STRATEGY_NETWORK_ONLY );
} );
```

This could be done for other extension points as well, such as managing navigation preload.

Maybe it would be more elegant way of managing these things, or maybe it is unnecessary and not as WordPressy.

Reply:
I think the filter approach is good for now. The templates can be modified through filters as well, and IMO it makes sense to continue along with it. This component is rather specific and not as generic as the ones for caching and precaching routes, which both furthermore have in common that they use registries. Last but not least, this component is hooked in to `wp_*_service_worker` at a very early priority (as it should), so relying on these same actions could result in problems due to hook priorities. I'm +1 for the filters.

One question I have though is, why do we only expose the filter for the frontend? Wouldn't it make sense in the admin too?

Reply:
> One question I have though is, why do we only expose the filter for the frontend? Wouldn't it make sense in the admin too?

Mainly because I wanted to keep the admin out of scope for now. Currently the offline and 500 error templates in the admin are hard-coded and cannot be customized. I was thinking that WordPress core would be responsible for managing how the admin would work as a PWA and that it wouldn't be in the realm of plugin/theme authors primarily to manage fundamentally.